src/django_aetos/views.py

```diff
@@ -18,12 +19,45 @@ def export_metrics(request):
                 metric["add_help"] = True
             known_metrics.add(metric["name_without_labels"])
             metrics.append(metric)
+    
+    return metrics
+
+def check_auth(request):
+    ip_address = request.META['REMOTE_ADDR']
+    allowed_ips = app_settings.IP_ALLOWLIST
+    
+    if app_settings.ENABLE_IP_ALLOWLIST and ip_address in allowed_ips:
+        return True
+    elif not app_settings.ENABLE_IP_ALLOWLIST:
+        return True
+    else:
+        return False
 
-    response = render(
-        request,
-        "metrics/export.txt",
-        context={"metrics": metrics},
-        content_type="text/plain",
-    )
-    response.content = re.sub(b"\n+", b"\n", response.content)
-    return response
+def check_ips(request):
+    try:
+        authorization_header = request.META['HTTP_AUTHORIZATION']
+    except KeyError:
+        authorization_header = ""
+    type, sep, token = authorization_header.partition(' ')
+    allowed_tokens = app_settings.AUTH_TOKENLIST
+    
+    if app_settings.ENABLE_AUTH and type == "Bearer" and token in allowed_tokens:
+        return True
+    elif not app_settings.ENABLE_AUTH:
+        return True
+    else:
+        return False
+
+def export_metrics(request):
+    if check_auth(request) and check_ips(request):
+        response = render(
+            request,
+            "metrics/export.txt",
+            context={"metrics": collect_response()},
+            content_type="text/plain",
+        )
+        response.content = re.sub(b"\n+", b"\n", response.content)
+        
+        return response
+    else:
+        return HttpResponse(status=401)
```

Comment:
Add a response here. At least "token or IP check failed". If we wan't to be nice, log which one failed.

Reply:
Differentiated response about which check failed 1ddd9c87f8bb9c78f7dc3998e406cf8b562178f4